### PR TITLE
Add update user endpoint

### PR DESCRIPTION
**Pull request by s-kakol**
* `src/common/utilities/swagger.config.ts` - bump docs version.
* `src/user/dto/update-user.dto.ts` - pretty much what can be updated, all fields are optional.
* `src/user/user.controller.ts` - PATCH endpoint for the update.
* `src/user/user.service.ts` - A bit messy. First we check if user even exists and we grab UserSettings. Then we try to update UserSettings and User. If we manage to do that then we return updated User, else we throw an error that might be thrown if user wants to change his email/username to one that's already taken or other undefined 400 error.

### CODE REVIEW

File: src/user/user.service.ts [hunk 1]
```diff
@@ -169,6 +170,45 @@ export class UserService {
       );
   }
 
+  async updateUser(id: number, dto: UpdateUserDto): Promise<UserEntity> {
+    await this.findUserById(id);
+    const userSettings = (await this.userSettingRepository
+      .createQueryBuilder('settings')
+      .where('settings.user_id = :id', { id })
```
Comment (adjurk) [inline src/user/user.service.ts:191]: I don't think I can see any logic related to verifying if the user who's submitted the request (based on their JWT token) is actually modifying their own settings and not anyone else's, which would pose a security vulnerability.

**Review by adjurk — CHANGES_REQUESTED** [review]
Let's address that one comment first and we should be good to go

File: src/user/user.service.ts (hunk 1 above)
Comment (s-kakol) [reply src/user/user.service.ts:191]: We've talked about this already. This is prevalent throughout the API and we agreed upon expanding the security if time allows.

Comment (s-kakol) [reply src/user/user.service.ts:191]: @adjurk If it bothers you I can switch what I was doing and implement it as I was looking for the best/quickest way to implement it anyway through the last few days and can have it done today/tomorrow as a part of a another task.

**Comment by s-kakol** [comment]
Commit: https://github.com/goodwill-ninjas/backend/pull/33/commits/412f8f7292f491507985ed6ef07175b71bc93455
* `src/user/dto/user-with-experience-details.dto.ts` - Add missing property `has_verified_email` to the DTO.
* `src/user/user.controller.ts` - Patch endpoint now also returns `UserWithExperienceDetails` instead of `UserEntity`. Why? Look service.
* `src/user/user.service.ts` - Because `updateUser()` method uses the `findUserById()` method that was changed in this branch to now return `UserWithExperienceDetails` instead of `UserEntity` I updated the return value of `updateUser()` from `UserEntity` to `UserWithExperienceDetails`.

**Review by adjurk — APPROVED** [review]

**Pull request merged**